src/lib/stitching/exchange/stitching.ts

```diff
@@ -248,6 +250,82 @@ export const exchangeStitchingEnvironment = ({
             })
           },
         },
+        artworkOrEditionSet: {
+          fragment: gql`
+            ... on CommerceLineItem {
+              artworkId
+              editionSetId
+            }
+          `,
+          resolve: async (parent, _args, context, info) => {
+            const artworkId = parent.artworkId
+            const editionSetId = parent.editionSetId
+
+            if (editionSetId) {
+              return info.mergeInfo.delegateToSchema({
+                schema: localSchema,
+                operation: "query",
+                fieldName: "artwork",
+                args: {
+                  id: artworkId,
+                },
+                context,
+                info,
+                transforms: [
+                  // Wrap document takes a subtree as an AST node
+                  new WrapQuery(
+                    // path at which to apply wrapping and extracting
+                    ["artwork"],
+                    (subtree: SelectionSetNode) => ({
+                      // we create a wrapping AST Field
+                      kind: Kind.FIELD,
+                      name: {
+                        kind: Kind.NAME,
+                        value: "editionSetById",
+                      },
+                      arguments: [
+                        {
+                          kind: Kind.ARGUMENT,
+                          name: {
+                            kind: Kind.NAME,
+                            value: "id",
+                          },
+                          value: {
+                            kind: Kind.STRING,
+                            value: editionSetId,
+                          },
+                        },
+                      ],
+                      // Inside the field selection
+                      selectionSet: subtree,
+                    }),
+                    // how to process the data result at path
+                    (result) => {
+                      return result.editionSetById
+                    }
+                  ),
+                ],
+              })
+            }
+
+            return info.mergeInfo
+              .delegateToSchema({
+                schema: localSchema,
+                operation: "query",
+                fieldName: "artwork",
+                args: {
+                  id: artworkId,
+                },
+                context,
+                info,
+                transforms: exchangeSchema.transforms,
+              })
+              .then((response) => {
+                response.__typename = "Artwork"
```

Comment:
This was getting coerced into `CommerceArtwork`, but we care about the first-class `Artwork` type here.

Reply:
This _might_ not be needed if you don't pass in the `transforms`. I think those aren't needed (since you aren't delegating to an exchange schema), and that might be what's pre-pending the `Commerce` part.